lib/chess/store/move.ex

```diff
@@ -0,0 +1,40 @@
+defmodule Chess.Store.Move do
+  @moduledoc false
+
+  use Ecto.Schema
+  use Timex.Ecto.Timestamps
+
+  import Ecto.Changeset
+
+  alias Chess.Store.Game
+
+  schema "moves" do
+    field :from, :map
+    field :to, :map
+
+    field :piece, :map
+    field :piece_captured, :map
+
+    belongs_to :game, Game
+
+    timestamps()
+  end
+
+  def changeset(struct, params \\ %{}) do
+    struct
+    |> cast(params, required_attrs())
+    |> validate_required(required_attrs())
+  end
+
+  def transform(move) do
+    %{
+      id: move.id,
+      piece: move.piece,
+      piece_captured: move.piece_captured,
+      from: <<97 + move.from["file"], 49 + move.from["rank"]>>,
```

Comment:
Can you explain what is going on here? I'm pretty lost.

Reply:
@drapergeek This is converting from zero indexed co-ordinates to a two character string/charlist by adding the co-ordinates to the ASCII code for `a` and `1` respectively. This gives us a charlist with a letter+number representation of the square co-ordinates.

```elixir
iex(1)> <<97 + 0, 49 + 0>>
"a1"
iex(5)> <<97 + 3, 49 + 6>>
"d7"
```